### PR TITLE
add workspace support

### PR DESCRIPTION
**Pull request by wexder**
This adds support for new go workspace file. It should behave the same as when using go modules.

### CODE REVIEW

**Review by edmorley — COMMENTED** [review]
(Switching review requests that need language-owner review to the owner directly, to prevent reminders the rest of the team can't action)

File: bin/compile [hunk 1]
```diff
@@ -533,6 +538,77 @@ case "${TOOL}" in
             bin/go-post-compile
         fi
     ;;
+    goworkspaces)
```
Comment (joshwlewis) [inline bin/compile:541]: This looks like an exact copy of the gomodules section? I can't easily tell what's different. Is there a way to reduce the duplication here? Maybe goworkspaces and gomodules could share the same branch of the case statement?

**Review by joshwlewis — COMMENTED** [review]
If you are still interested in pursuing this PR, it'll need some tests of the new functionality.

**Comment by wexder** [comment]
> If you are still interested in pursuing this PR, it'll need some tests of the new functionality.

I'm no longer working with heroku so I'm not interested in finishing this. Anyone is free to pick up the work where I left it, but It looks like noone needed this is last 2 years...

**Pull request closed without merging**